### ISSUE TITLE
Instrumentation Error with .NET Core 3.1

### ISSUE THREAD

**Issue by PSanetra** [SteveGilham/altcover#87]
Altcover does not seem to work with .NET Core 3.1 (haven't tried with older versions). It fails during the instrumentation phase with the following error:

```
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(41,5): error :  [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(41,5): error : ERROR *** Instrumentation phase failed [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(41,5): error :  [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(41,5): error :  [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(41,5): error : Could not load file or assembly 'Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. Could not find or load a specific file. (0x80131621) [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(41,5): error : Details written to /src/SimpleAppTests/bin/Debug/netcoreapp3.1/__Saved/AltCover-2020-06-29--09-33-52.log [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(41,5): error : If this problem was detected in the pre-test instrumentation stage of `dotnet test`, then the file may have been moved to /src/SimpleAppTests/bin/Debug/netcoreapp3.1/AltCover-2020-06-29--09-33-52.log when the task completes. [/src/SimpleAppTests/SimpleAppTests.csproj]
Test run for /src/SimpleAppTests/bin/Debug/netcoreapp3.1/SimpleAppTests.dll(.NETCoreApp,Version=v3.1)
Microsoft (R) Test Execution Command Line Tool Version 16.3.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...

A total of 1 test files matched the specified pattern.

Test Run Successful.
Total tests: 1
     Passed: 1
 Total time: 0.9339 Seconds
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(125,5): warning : A total of 0 visits recorded [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(125,5): error MSB4018: The "AltCover.Collect" task failed unexpectedly. [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(125,5): error MSB4018: System.NullReferenceException: Object reference not set to an instance of an object. [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(125,5): error MSB4018:    at AltCover.Zip.openUpdate(String report) in C:\projects\altcover\AltCover\CommandLine.fs:line 79 [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(125,5): error MSB4018:    at AltCover.Runner.J.writeReportBase@823.Invoke(FSharpOption`1 arg) in C:\projects\altcover\AltCover\Runner.fs:line 823 [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(125,5): error MSB4018:    at AltCover.Runner.value@877-3.Invoke(Unit unitVar0) in C:\projects\altcover\AltCover\Runner.fs:line 893 [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(125,5): error MSB4018:    at AltCover.CommandLine.I.doPathOperation[a](FSharpFunc`2 f, a defaultValue, Boolean store) in C:\projects\altcover\AltCover\CommandLine.fs:line 210 [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(125,5): error MSB4018:    at AltCover.CommandLine.doPathOperation@395.Invoke(FSharpFunc`2 f, a defaultValue, Boolean store) in C:\projects\altcover\AltCover\CommandLine.fs:line 395 [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(125,5): error MSB4018:    at AltCover.Runner.doCoverage(String[] arguments, OptionSet options1) in C:\projects\altcover\AltCover\Runner.fs:line 876 [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(125,5): error MSB4018:    at AltCover.Main.I.main(String[] arguments) in C:\projects\altcover\AltCover\Main.fs:line 525 [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(125,5): error MSB4018:    at AltCover.Main.effectiveMain@531.Invoke(String[] arguments) in C:\projects\altcover\AltCover\Main.fs:line 531 [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(125,5): error MSB4018:    at AltCover.Collect.Execute() in C:\projects\altcover\AltCover\Tasks.fs:line 226 [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(125,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/src/SimpleAppTests/SimpleAppTests.csproj]
/root/.nuget/packages/altcover/7.1.776/build/netstandard2.0/AltCover.targets(125,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/src/SimpleAppTests/SimpleAppTests.csproj]
The command '/bin/sh -c dotnet test /p:AltCover=true /p:AltCoverCobertura=coverage.xml /p:CopyLocalLockFileAssemblies="true"' returned a non-zero code: 1
```

I have set up a repository, which can be used to reproduce the error with a Docker image build: https://github.com/PSanetra/reproduce-altcover-error

**Comment by SteveGilham** [comment]
Could not repro with a freshly created docker machine.

![2020-06-19](https://user-images.githubusercontent.com/7460854/86007948-0e470580-ba10-11ea-90f5-f2eeff624e56.png)

The message
```
Could not load file or assembly 'Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. Could not find or load a specific file. (0x80131621)
```
would appear to indicate some failure or corruption in the installation of the NuGet package, as that assembly forms part of the package manifest.

**Comment by PSanetra** [comment]
@SteveGilham ok, weird. I am currently also not able to reproduce this issue in a GitHub Actions pipeline. We could reproduce the issue on two different systems (ubuntu and mac os) with a different repository. On my local machine this issue is also appearing in the docker build of the example repository.

We have also discovered that we can not reproduce the issue with version 6.8.761 of altcover.

**Comment by SteveGilham** [comment]
There was some issue (can't recall details at this remove) I had when rolling out the .net core support a couple of years ago, where Newtonsoft.Json was implicated the cause of build (strictly, the whole build-and-validate process) failures.  I finally resolved that problem by static-linking that assembly into the main executable.

The .net 3.1.300 SDK had [a regression which affects that static linkage](https://github.com/dotnet/fsharp/issues/9255), so the 7.x releases have been built without that step, but have seemed to function OK without it.  Release 6.8.761 is the last that was done with an older .net SDK and thus with the static linking happening; so if there is a re-emergence of the original problem I had, whatever it was, or something very like, then sticking with a 6.x version would avoid it.

**Comment by SteveGilham** [comment]
[Release 7.1.778](https://www.nuget.org/packages/AltCover/7.1.778) replaces the Newtonsoft assembly that is at the root of the issue, plus tidies up the downstream failures.

**Comment by PSanetra** [comment]
@SteveGilham I can confirm that this issue is fixed with version 7.1.778
